packages/palette/src/elements/Select/Select.tsx

```diff
@@ -30,102 +36,114 @@ export interface SelectProps
+      selected,
+      title,
+      variant = "default",
+      onSelect,
+      ...rest
+    },
+    forwardedRef
+  ) => {
+    const ref = useRef<HTMLSelectElement | null>(null)
+
+    const [boxProps, selectProps] = splitBoxProps(rest)
+
+    return (
+      <Box width="100%" {...boxProps}>
+        <Flex
+          as="label"
+          {...(variant === "inline"
+            ? {
+                flexDirection: "row",
+                alignItems: "center",
               }
-            >
-              {title}
-              {required && (
-                <Box as="span" color="brand">
-                  *
-                </Box>
-              )}
-            </Text>
-          )}
-
-          {description && (
-            <Text variant="xs" color="black60">
-              {description}
-            </Text>
-          )}
-        </div>
-
-        <Container
-          variant={variant}
-          disabled={!!disabled}
-          hover={!!hover}
-          error={error!}
-          focus={!!focus}
-          mt={variant !== "inline" && (title || description) ? 0.5 : 0}
+            : {
+                flexDirection: "column",
+                alignItems: "flex-start",
+              })}
+          {...(id ? { for: id } : {})}
         >
-          <select
-            id={id}
-            disabled={disabled}
-            name={name}
-            value={selected}
-            onChange={(event) => {
-              onSelect && onSelect(event.target.value)
-            }}
-            {...selectProps}
+          <div>
+            {title && (
+              <Text
+                variant="xs"
+                lineHeight={
+                  variant === "inline" && description === undefined
+                    ? 1
+                    : undefined
+                }
+              >
+                {title}
+                {required && (
+                  <Box as="span" color="brand">
+                    *
+                  </Box>
+                )}
+              </Text>
+            )}
+
+            {description && (
+              <Text variant="xs" color="black60">
+                {description}
+              </Text>
+            )}
+          </div>
+
+          <Container
+            variant={variant}
+            disabled={!!disabled}
+            hover={!!hover}
+            error={error!}
+            focus={!!focus}
+            mt={variant !== "inline" && (title || description) ? 0.5 : 0}
           >
-            {options.map(({ value, text }) => {
-              return (
-                <option value={value} key={value}>
-                  {text}
-                </option>
-              )
-            })}
-          </select>
-        </Container>
-      </Flex>
-
-      {error && typeof error === "string" && (
-        <Text variant="xs" mt={0.5} color="red100">
-          {error}
-        </Text>
-      )}
-    </Box>
-  )
-}
+            <select
+              ref={composeRefs(ref, forwardedRef) as any}
```

Comment:
You won't need this compose here; just pass the forwarded ref.

Reply:
fixed